src/lib/command-helpers.mjs

```diff
@@ -66,15 +98,99 @@ export const commonConfigurableQueryOptions = {
 };
 
 /**
- * Validate that the user has specified either a database or a secret.
- * This check is not required for commands that can operate at a
- * "root" level.
- * @param {object} argv 
- * @param {string} argv.database - The database to use
- * @param {string} argv.secret - The secret to use
+ * This function will return a v4 or v10 client based on the version provided in the argv.
+ *   onInvalidCreds decides whether or not we retry or ask the user to re-enter their secret.
+ * @param {*} argv
+ * @returns { Promise<any> } - A Fauna client
  */
-export const validateDatabaseOrSecret = (argv) => {
-  if (!argv.database && !argv.secret) {
-    throw new Error("No database or secret specified. Pass --database or --secret.");
+export async function getSimpleClient(argv) {
+  const logger = container.resolve("logger");
+  const credentials = container.resolve("credentials");
+  let client = await buildClient(argv);
+  const originalQuery = client.query.bind(client);
+
+  const queryArgs = async (originalArgs) => {
+    const queryValue = originalArgs[0];
+    const queryOptions = {
+      ...originalArgs[1],
+      secret: await credentials.databaseKeys.getOrRefreshKey(),
+    };
+    return [queryValue, queryOptions];
+  };
+
+  client.query = async function (...args) {
+    const updatedArgs = await queryArgs(args);
+    return originalQuery(...updatedArgs).then(async (result) => {
+      if (result.status === 401) {
+        // Either refresh the db key or tell the user their provided key was bad
+        logger.debug(
+          "Invalid credentials for Fauna API Call, attempting to refresh",
+          "creds",
+        );
+        await credentials.databaseKeys.onInvalidCreds();
+        const updatedArgs = await queryArgs(args);
+        return await originalQuery(...updatedArgs);
+      }
+      return result;
+    });
+  };
+
+  return client;
```

Comment:
Do we need to return a monkey-patched version of the client which could be v4 or v10? Or can we return a new POJO with a `query` property?

Reply:
this is a rebasing error. this method is not used. i will remove it.